src/common/catalog-entities/kubernetes-cluster.ts

```diff
@@ -19,6 +20,38 @@ export class KubernetesCluster extends CatalogEntity<CatalogEntityMetadata, Kube
   public readonly apiVersion = "entity.k8slens.dev/v1alpha1";
   public readonly kind = "KubernetesCluster";
 
+  async connect(): Promise<boolean> {
+    if (app) {
```

Comment:
Is there a use case for this being run on `main` at all? I guess that should be discussed in the future. But it seems very strange indeed for any `CatalogEntity` on `main` providing a function something like `onContextMenuOpen` or even `connect` for that matter.

Reply:
Use case for `main`, currently no. It just feels old cluster class all over again if it works partially on the other side.

Reply:
I don't understand:

> It just feels old cluster class all over again

but I think this is a better version.

Reply:
Lens 4.x `Cluster` object had public methods that worked only in `main`.